src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -93,6 +93,9 @@ c_static_assert(PAL_S_IFSOCK == S_IFSOCK);
 // Validate that our enum for inode types is the same as what is
 // declared by the dirent.h header on the local system.
 // (AIX doesn't have dirent d_type, so none of this there)
+#if !defined (__clang__) && defined (__GNUC__)
```

Comment:
One last nit - it seems we could get rid of the warning by casting both sides of the comparison to `int` to get rid of the need for this pragma.

Reply:
I have removed the suppression and used the cast approach.

One thing, I just realized that this PR (as of fd86d29) is only fixing the debug build for coreclr, installer and libraries native components. However, if we build with `-release`, there are more things to fix. I will try to address those separately.